assessment/libs/finderiv.php

```diff
@@ -0,0 +1,537 @@
+// Author:  Daniel Brown
+//          University of Colorado Boulder
+
+global $allowedmacros;
+array_push($allowedmacros,"finderiv_payout","finderiv_fwdprice","finderiv_fwdpricediv",
+	"finderiv_fwdcontract",	"finderiv_bsm","finderiv_immdate","finderiv_equityfutdate", 
+	"finderiv_convertrate","finderiv_fairforwardrate","finderiv_fra");
+
+// function finderiv_payout
+// calculates the payout of a portfolio given a terminal value for the 
+// underlying asset. The portfolio can contain
+// European calls, European puts, zero coupon bonds and forward contracts
+function finderiv_payout($asset, $types, $possizes, $strikes){
+	if (!is_numeric($asset) ) {
+		echo "finderiv_payout: invalid asset level";
+		return false;
+	}
+	if (!is_array($types)) {
+		echo 'finderiv_payout: types must be an array';
+		return false;
+	}
+	$numTypes = count($types);
+	if (!is_array($possizes)) {
+		echo 'finderiv_payout: position sizes must be an array';
+		return false;
+	}
+	$numpossizes = count($possizes);
+	if (!is_array($strikes)) {
+		echo 'finderiv_payout: strikes must be an array';
+		return false;
+	}
+	$numStrikes = count($strikes);
+	if ($numTypes!=$numpossizes) {
+		echo 'finderiv_payout: type and position size arrays must be of the same size.';
+		return false;
+	}
+	if ($numTypes!=$numStrikes) {
+		echo 'finderiv_payout: type and strike arrays must be of the same size.';
+		return false;
+	}
+	$total = 0;
+    for ($i=0;$i<$numTypes;$i++) {
+		if (!is_numeric($possizes[$i])) {
+			echo "finderiv_payout: position size is invalid. Cannot be " . $possizes[$i] . ".\r\n";
+			return false;
+		}
+		if (!is_numeric($strikes[$i])) {
+			echo "finderiv_payout: strike is invalid. Cannot be " . $strikes[$i] . ".";
+			return false;
+		}
+
+		if (strcasecmp($types[$i],"call")==0) {
+			$total += $possizes[$i]* max($asset-$strikes[$i],0);
+		} elseif (strcasecmp($types[$i],"put")==0) {
+			$total += $possizes[$i]* max($strikes[$i]-$asset,0);
+		} elseif (strcasecmp($types[$i],"bond")==0) {
+			$total += $possizes[$i];
+		} elseif (strcasecmp($types[$i],"forward")==0) {
+			$total += $possizes[$i]*($asset-$strikes[$i]);
+		} else {
+			echo 'finderiv_payout: type ' . $types[$i] . ' is not recognized. It must be call, put, forward, bond';
+			return false;
+		}
+	}
+	return $total;
+}
+
+// function finderiv_fwdprice
+// This function returns the fair forward price for an 
+// asset. It handles the situtation where the asset earns a yield and thus 
+// can be used for equity indicies or foreign exchange
+// mat is given in years
+function finderiv_fwdprice($spot, $rf, $yield, $today,$mat,$fmt="F j, Y") {
+	if (!is_numeric($spot) ) {
+		echo "finderiv_fwdprice: invalid spot price";
+		return false;
+	}
+	if (!is_numeric($rf) ) {
+		echo "finderiv_fwdprice: invalid risk-free interest rate";
+		return false;
+	}
+	if (!is_numeric($yield) ) {
+		echo "finderiv_fwdprice: invalid yield";
+		return false;
+	}
+
+	$todaydate = date_create_from_Format($fmt,$today);
```

Comment:
This should be `date_create_from_format`, no cap on the Format.  Same everywhere else this function is used.

Reply:
Sorry for the confusion.  I have corrected this.